webapp/src/Service/ExternalContestSourceService.php

```diff
@@ -735,30 +695,30 @@ protected function validateAndUpdateContest(string $entityType, ?string $eventId
 
         // We need to convert the freeze to a value from the start instead of
         // the end so perform some regex magic.
-        $duration     = $data['duration'];
-        $freeze       = $data['scoreboard_freeze_duration'];
+        $duration = $data->duration;
+        $freeze = $data->scoreboardFreezeDuration;
         $reltimeRegex = '/^(-)?(\d+):(\d{2}):(\d{2})(?:\.(\d{3}))?$/';
         preg_match($reltimeRegex, $duration, $durationData);
 
         $durationNegative = ($durationData[1] === '-');
-        $fullDuration     = $durationNegative ? $duration : ('+' . $duration);
+        $fullDuration = $durationNegative ? $duration : ('+' . $duration);
 
         if ($freeze !== null) {
             preg_match($reltimeRegex, $freeze, $freezeData);
-            $freezeNegative       = ($freezeData[1] === '-');
-            $freezeHourModifier   = $freezeNegative ? -1 : 1;
-            $freezeInSeconds      = $freezeHourModifier * (int)$freezeData[2] * 3600
-                                    + 60 * (int)$freezeData[3]
-                                    + (double)sprintf('%d.%03d', $freezeData[4], $freezeData[5]);
+            $freezeNegative = ($freezeData[1] === '-');
+            $freezeHourModifier = $freezeNegative ? -1 : 1;
+            $freezeInSeconds = $freezeHourModifier * (int)$freezeData[2] * 3600
+                + 60 * (int)$freezeData[3]
+                + (double)sprintf('%d.%03d', $freezeData[4], $freezeData[5] ?? 0);
             $durationHourModifier = $durationNegative ? -1 : 1;
-            $durationInSeconds    = $durationHourModifier * (int)$durationData[2] * 3600
-                                    + 60 * (int)$durationData[3]
-                                    + (double)sprintf('%d.%03d', $durationData[4], $durationData[5]);
-            $freezeStartSeconds   = $durationInSeconds - $freezeInSeconds;
-            $freezeHour           = floor($freezeStartSeconds / 3600);
-            $freezeMinutes        = floor(($freezeStartSeconds % 3600) / 60);
-            $freezeSeconds        = floor(($freezeStartSeconds % 60) / 60);
-            $freezeMilliseconds   = $freezeStartSeconds - floor($freezeStartSeconds);
+            $durationInSeconds = $durationHourModifier * (int)$durationData[2] * 3600
+                + 60 * (int)$durationData[3]
+                + (double)sprintf('%d.%03d', $durationData[4], $durationData[5] ?? 0);
+            $freezeStartSeconds = $durationInSeconds - $freezeInSeconds;
+            $freezeHour = floor($freezeStartSeconds / 3600);
+            $freezeMinutes = floor(($freezeStartSeconds % 3600) / 60);
+            $freezeSeconds = floor(($freezeStartSeconds % 60) / 60);
+            $freezeMilliseconds = $freezeStartSeconds - floor($freezeStartSeconds);
```

Comment:
Please don't just reformat code if you're not touching it, and if you do, then do it in separate commits. Also, I disagree with this kind of removal of alignment spaces. It makes the code less readable.

Reply:
I think the reformatting was done in Nicky his PR.

I added a change to his commits but this branch is (currently) mostly to make the shadowing work for tonight.

Reply:
Yes, I found out afterwards, so let's have that discussion there.

As a temporary I think this is fine, but if this planned to be merged into main, then please rebase it on top of Nicky's merged branch, so that only the actual changes are shown in the PR.